### PR TITLE
Theone keyboard

### PR DESCRIPTION
**Pull request by txdyqiang**
We submitted a code for "the one" custom keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by adophoxia** [comment]
I'd hope this is the only PR for this board cause if this were to be closed and a new one is opened that is the exact same as this one, then you'd be subjected to being banned from the repo.

**Comment by txdyqiang** [comment]
> I'd hope this is the only PR for this board cause if this were to be closed and a new one is opened that is the exact same as this one, then you'd be subjected to being banned from the repo.
Sorry, I promise this won’t happen again.

**Review by waffle87 — COMMENTED** [review]
There are numerous things wrong
* Missing top level `rules.mk` file — firmware will not compile
* Missing RGB Matrix configuration in `info.json`
* This seems to be for an entirely different keyboard? (`"keyboard_name": "NuPhy Krypton75"`)

File: keyboards/theone/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2023 Persama (@Persama)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+// layer Mac
+[0] = LAYOUT(
+	KC_ESC, 	KC_BRID,  	KC_BRIU,  	_______, 	_______, 	_______,  	_______,  	KC_MPRV,  	KC_MPLY,  	KC_MNXT, 	KC_MUTE, 	KC_VOLD, 	KC_VOLU, 				KC_DEL,
```
Comment (fauxpark) [inline keyboards/theone/keymaps/default/keymap.c:9]: Transparent keys on layer 0 makes no sense.

File: keyboards/theone/config.h
Comment (fauxpark) [inline keyboards/theone/config.h:1]: Most of these can be moved to info.json.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/theone/keymaps/default/keymap.c (hunk 1 above)
Comment (txdyqiang) [reply keyboards/theone/keymaps/default/keymap.c]: Thanks，I have replaced Transparent keys with F keys.

File: keyboards/theone/config.h
Comment (txdyqiang) [reply keyboards/theone/config.h:1]: Thanks, These have been moved to info.json.

File: keyboards/theone/info.json
[the remark this answers is not in the document]
Comment (txdyqiang) [reply keyboards/theone/info.json:1]: This file has been edited by running the command.

File: keyboards/theone/info.json [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+            {"matrix": [5, 0], "x": 0, "y": 50, "flags": 4}
+        ],
+        "max_brightness": 130,
+        "speed_steps": 52,
+        "val_steps": 26
+    },
+    "usb": {
+        "device_version": "0.0.1",
+        "pid": "0x3245",
+        "vid": "0x19F5"
```
Comment (drashna) [inline keyboards/theone/info.json:159]: This Vendor ID is in use: 
https://www.the-sz.com/products/usbid/index.php?v=0x19F5&p=&n=

Generally, above 0x4000 is best to use, and the further past it, the better.

File: keyboards/theone/config.h [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2023 Persama (@Persama)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma once
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 8  //  This is 4 by default.
```
Comment (drashna) [inline keyboards/theone/config.h:5]:
```suggestion
```

File: keyboards/theone/config.h [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+#define EECONFIG_USER_DATA_SIZE    8  // This is the size of the EEPROM for the custom VIA-specific data
+
+#define RGB_MATRIX_LED_COUNT       89
+
+
+#define RGB_MATRIX_DEFAULT_MODE       RGB_MATRIX_CYCLE_LEFT_RIGHT  // Sets the default mode, if none has been set
+#define RGB_DISABLE_WHEN_USB_SUSPENDED                             // turn off effects when suspended
```
Comment (drashna) [inline keyboards/theone/config.h:13]: cycle left/right is the default mode .... by default.  No need to set it.
```suggestion
```

File: keyboards/theone/info.json [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+            {"matrix": [5, 12], "x": 120, "y": 50, "flags": 4},
+            {"matrix": [5, 10], "x": 110, "y": 50, "flags": 4},
+            {"matrix": [5, 9], "x": 100, "y": 50, "flags": 4},
+            {"matrix": [5, 6], "x": 37.5, "y": 50, "flags": 4},
+            {"matrix": [5, 2], "x": 25, "y": 50, "flags": 4},
+            {"matrix": [5, 1], "x": 12.5, "y": 50, "flags": 4},
+            {"matrix": [5, 0], "x": 0, "y": 50, "flags": 4}
+        ],
+        "max_brightness": 130,
+        "speed_steps": 52,
```
Comment (drashna) [inline keyboards/theone/info.json:153]:
```suggestion
        "sleep": true,
        "speed_steps": 52,
```

File: keyboards/theone/info.json [hunk 6]
```diff
@@ -0,0 +1,251 @@
+{
+    "manufacturer": "github.com/sowbug",
+    "keyboard_name": "Theone",
+    "bootloader": "stm32-dfu",
+    "diode_direction": "COL2ROW",
+    "features": {
```
Comment (drashna) [inline keyboards/theone/info.json:6]:
```suggestion
    "dynamic_keymap": {
        "layer_count": 8
    },
    "features": {
```

File: keyboards/theone/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/theone/keymaps/default/keymap.c:1]: please replace all of the tabs with spaces in this file.

File: keyboards/theone/keymaps/via/keymap.c
Comment (drashna) [inline keyboards/theone/keymaps/via/keymap.c:1]: Please replace all of the tabs with spaces.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/theone/keymaps/default/keymap.c
Comment (txdyqiang) [reply keyboards/theone/keymaps/default/keymap.c:1]: Thanks for your comments, I've replaced all of the tabs with spaces.

File: keyboards/theone/keymaps/via/keymap.c
Comment (txdyqiang) [reply keyboards/theone/keymaps/via/keymap.c:1]: Thanks for your comments, I've replaced all of the tabs with spaces.

File: keyboards/theone/info.json (hunk 2 above)
Comment (txdyqiang) [reply keyboards/theone/info.json]: Thanks, the Vendor ID has been edited to 0x49F5.

File: keyboards/theone/info.json [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+            {"matrix": [5, 9], "x": 100, "y": 50, "flags": 4},
+            {"matrix": [5, 6], "x": 37.5, "y": 50, "flags": 4},
+            {"matrix": [5, 2], "x": 25, "y": 50, "flags": 4},
+            {"matrix": [5, 1], "x": 12.5, "y": 50, "flags": 4},
+            {"matrix": [5, 0], "x": 0, "y": 50, "flags": 4}
+        ],
+        "max_brightness": 130,
+        "sleep": true,
+        "speed_steps": 52,
+        "val_steps": 26
```
Comment (zvecr) [inline keyboards/theone/info.json:158]: These values are poor choices as default values.
```suggestion
        "sleep": true
```

File: keyboards/theone/info.json [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+            "solid_reactive_simple": true,
+            "solid_reactive_wide": true,
+            "solid_splash": true,
+            "splash": true,
+            "typing_heatmap": true
+        },
+        "center_point": [112, 30],
+        "driver": "ws2812",
+        "layout": [
+            {"matrix": [0, 0], "x": 0, "y": 0, "flags": 4},
```
Comment (zvecr) [inline keyboards/theone/info.json:73]: You have 81 elements here, but declare 89 LEDs.

File: keyboards/theone/config.h
Comment (zvecr) [inline keyboards/theone/config.h:1]: This file should be removed,

* RGB_MATRIX_LED_COUNT is inferred from `len(rgb_matrix.layout)` in info.json
* EECONFIG_USER_DATA_SIZE isnt required

**Review by zvecr — CHANGES_REQUESTED** [review]
Also, `readme.md` (following [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md)) is missing

File: keyboards/theone/config.h
Comment (txdyqiang) [reply keyboards/theone/config.h:1]: Thanks for the reminder.
I calculated an incorrect value of  RGB_MATRIX_LED_COUNT.
As your suggestion, I removed RGB_MATRIX_LED_COUNT and EECONFIG_USER_DATA_SIZE from info.json

Comment (txdyqiang) [reply keyboards/theone/config.h:1]: The file has been removed, thank you again for your suggestion.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**